.travis.yml

```diff
@@ -5,14 +5,27 @@ cache:
   - $HOME/.m2
 language: java
 jdk:
-- oraclejdk7
-services: mongodb
+  - oraclejdk8
 install:
-- ./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V -U
+  - wget http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-${MONGODB}.tgz
+  - tar zxf mongodb-linux-x86_64-${MONGODB}.tgz
+  - ${PWD}/mongodb-linux-x86_64-${MONGODB}/bin/mongod --version
+  - ./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V -U
+before_script:
+  - mkdir ${PWD}/mongodb-linux-x86_64-${MONGODB}/data
+  - ${PWD}/mongodb-linux-x86_64-${MONGODB}/bin/mongod --dbpath ${PWD}/mongodb-linux-x86_64-${MONGODB}/data --logpath ${PWD}/mongodb-linux-x86_64-${MONGODB}/mongodb.log --fork
+after_script:
+  - pkill mongod
 script:
-- '[ "${TRAVIS_BRANCH}" = master ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ] || ./mvnw -s .settings.xml clean verify deploy'
-- '[ "${TRAVIS_PULL_REQUEST}" = "false" ] || ./mvnw clean verify'
+  - '[ "${TRAVIS_BRANCH}" = master ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ] || ./mvnw -s .settings.xml clean verify deploy'
+  - '[ "${TRAVIS_PULL_REQUEST}" = "false" ] || ./mvnw clean verify'
 env:
+  matrix: 
+    - MONGODB=3.2.20
+    - MONGODB=3.4.16
+    - MONGODB=3.6.6
+    - MONGODB=4.0.0
+    - MONGODB=4.1.1
```

Comment:
I'd drop `MONGODB=4.1.1` as I don't find any release notes for it, and I'm unsure what license it's released under.